pkg/applicationmanager/controllers/applicationbackup.go

```diff
@@ -2226,6 +2226,53 @@ func (a *ApplicationBackupController) cleanupResources(
 		log.ApplicationBackupLog(backup).Errorf("%v", errMsg)
 		return err
 	}
+
+	if IsBackupObjectTypeVirtualMachine(backup) {
```

Comment:
Should we also check whether SkipAutoExecRules is not set to entry here? I mean if SkipAutoExecRules is set, we will not be deleted here rt?

Reply:
Also I feel that @sgajawada-px code change will try to delete the VM rule as well rt as he is checking for created-by from the applicationbackup CR.

Reply:
Yes add && SkipVmAutoExecRule flag to this if

Reply:
@sgajawada-px code wont delete the rule as he checks for 
 create-by value to utils.PxbackupAnnotationCreateByValue  which is set to px-backup
while we set the create-by value to stork..

Reply:
@sgajawada-px is taking the CreateByValue from the applicationbackup cluster not from Rules CR. I think you are setting the createdBYValue to strok in rule CR not in AB CR. Is that not correct?